api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsExpirationServiceImpl.java

```diff
@@ -86,14 +86,16 @@ && haveCreditsExpired(user)
           "Initial credits expired for user {}. Expiration time: {}",
           user.getUsername(),
           userInitialCreditsExpiration.getExpirationTime());
+      boolean areInitialCreditsExhausted =
```

Comment:
In order to avoid recalculating whether a user has exhausted their credits, I am checking to see if any of the workspaces that they own are exhausted in order to make sure that they are not receiving two similar emails.

The parallelStream.anyMatch is a short circuit approach to see if any of the workspaces are reporting as exhausted:
https://stackoverflow.com/a/26305019/2166475

Reply:
I don't think that I've used parallelStream before, but if stream can do this too, wouldn't parallelStream be faster, because it can evaluate multiple workspaces at the same time via multi-threading?

I am new to this sort of thing, so I could definitely be missing something.

Reply:
I haven't thought a lot about the pro/con of stream vs. parallelStream here.  What you're saying makes sense.  My default is for non-threaded because it seems simpler to understand.